storage/remote/generic/generic.go

```diff
@@ -62,16 +69,15 @@ func (c *Client) Store(samples model.Samples) error {
 		}
 		req.Timeseries = append(req.Timeseries, ts)
 	}
-	client := NewGenericWriteClient(c.conn)
 	ctxt, _ := context.WithTimeout(context.Background(), c.timeout)
```

Comment:
The client should probably be created with a base context from which the request contexts are derived from.